homeassistant/components/generic_thermostat/climate.py

```diff
@@ -266,20 +266,21 @@ def _async_startup(*_):
 
         # Check If we have an old state
         if (old_state := await self.async_get_last_state()) is not None:
-            # If we have no initial temperature, restore
```

Comment:
The initial state should always overrule a restored state. If the user wants it to restore, they should not set an initial temperature.